src/Network/CurlResult.php

```diff
@@ -226,11 +226,26 @@ public function getContentType()
 	/**
 	 * Returns the Curl headers
 	 *
-	 * @return string the Curl headers
+	 * @param string $field optional header field. Return all fields if empty
+	 *
+	 * @return string|bool the Curl headers, "false" when field isn't found
```

Comment:
I'm not a fan of mixed types in returns, but I think it's unfortunately the easiest way this time :\

Reply:
Yeah, it's the way that many PHP functions do work.

Reply:
Yes, but these PHP functions are neither a class instance, nor do they have a state. This method is a getter of a class instance, so headers are strings. Because of these types, it's not logic to return other types than the type of the field behind the getter. Normally I' d say either return empty and the logic of the caller is responsible for handling it or we should implement a `hasHeader()` method for the Curl class, which checks if the field is possible. I'd prefer the second solution

Reply:
I really don't like using a second function. This drastically reduces the usability.

Reply:
But it makes the usage and expectation after the call more clear.. What if the value of the header field is empty and you pass a empty parameter. The return value is empty which can be seen as false in case it's missing or its false because the value is empty. That are different meanings of the same return..

Reply:
With one function you have to write either
```php
if ($CurlResult->getHeader('Accept') !== false) {
    $accept = $CurlResult->getHeader('Accept');
    ...
}
```
or
```
$accept = $CurlResult->getHeader('Accept');
if ($accept !== false) {
    ...
}
```
And both aren't that much more usable. In one case you call the same function twice, in the second you end up with an intermediary value you may not have any use for.

This form is better readable and ultimately, usable:
```php
if ($CurlResult->hasHeader('Accept')) {
    $accept = $CurlResult->getHeader('Accept');
    ...
}
```